packages/orm/src/relations/ManyToOneReference.ts

```diff
@@ -89,14 +93,23 @@ export class ManyToOneReference<T extends Entity, U extends Entity, N extends ne
     return this.doGet({ withDeleted: false });
   }
 
-  get id(): IdOf<U> | undefined {
+  get id(): IdOf<U> | N {
     ensureNotDeleted(this.entity, { ignore: "pending" });
-    return maybeResolveReferenceToId(this.current()) as IdOf<U> | undefined;
+    return maybeResolveReferenceToId(this.current()) as IdOf<U> | N;
+  }
+
+  set id(id: IdOf<U> | N) {
```

Comment:
This is new. Technically our `Reference.id` was not previously marked as `readonly`, but kinda surprising, TS didn't complain that we didn't have a setter :thinking: :shrug: